plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/IStreamValueEditorPersistent.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ui.data;
+
+import org.eclipse.swt.widgets.Control;
+import org.jkiss.code.NotNull;
+import org.jkiss.code.Nullable;
+
+import java.nio.file.Path;
+
+public interface IStreamValueEditorPersistent <CONTROL extends Control> {
+
+    /**
+     * @param control Control to acquire filepath from
+     * @return filepath of external file or null if not found
+     */
```

Comment:
This is a bad Javadoc without a body.

The `control` is not used to acquire the file path. It can be obtained from anywhere. It does not strictly need to be the `control` which provides the file path. It's just convenience since the editor itself doesn't know its control. TL;DR: call it `editor control` or something like that